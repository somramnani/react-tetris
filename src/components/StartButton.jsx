import { StyledStartButton } from "./styles/StyledStartButton";

const StartButton = ({ callback }) => {
  return (
    <StyledStartButton data-testid="startButton-1" onClick={callback}>
      Start Game
    </StyledStartButton>
  );
};

export default StartButton;
