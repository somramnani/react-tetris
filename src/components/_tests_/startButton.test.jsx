import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import StartButton from "../StartButton";

describe("Start Button Component", () => {
  let mockCallback;

  afterEach(() => {
    mockCallback = jest.fn();
  });

  it("should render Start Button component", () => {
    render(<StartButton callback={mockCallback} />);

    const startButtonElement = screen.getByTestId("startButton-1");

    expect(startButtonElement).toBeInTheDocument();
    expect(startButtonElement).toHaveTextContent("Start Game");
  });

  it("calls the callback when button is clicked", () => {
    render(<StartButton callback={mockCallback} />);

    fireEvent.click(screen.getByTestId("startButton-1"));

    expect(mockCallback).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<StartButton callback={mockCallback} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
