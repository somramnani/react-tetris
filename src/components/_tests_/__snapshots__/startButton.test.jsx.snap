// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start Button Component should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-aYaIB fbOjTh"
    data-testid="startButton-1"
  >
    Start Game
  </button>
</DocumentFragment>
`;
